src/Cloneable.php

```diff
@@ -11,7 +11,9 @@ public function with(...$values): static
         $clone = (new ReflectionClass(static::class))->newInstanceWithoutConstructor();
 
         foreach ($this as $objectField => $objectValue) {
-            $objectValue = $values[$objectField] ?? $objectValue;
+            if (array_key_exists($objectField, $values)) {
```

Comment:
This refactor doesn't seem to have any benefit. I think the previous code is readable enough.

Reply:
The previous code does not work in all situation. I added a test to prove it.
So this is not a refactor, this is a bug fix 😎